### ISSUE TITLE
admin: SSP SP-metadata doesn't contain 'entityid' key

### ISSUE THREAD

**Issue by tvdijen** [simplesamlphp/simplesamlphp#1805]
**Describe the bug**
The SP-metadata (SSP-array style) does not contain the 'entityid' key.
This is problematic, because some code depends on it (i.e. https://github.com/simplesamlphp/simplesamlphp/blob/master/src/SimpleSAML/Metadata/MetaDataStorageHandler.php#L377), effectively breaking HTTP-Artifact binding and possibly more.

**To Reproduce**
1. Set up SSP as a service provider
2. Go to admin Federation page and pull down the SSP metadata
3. See no 'entityid' key

**Expected behavior**
An 'entityid' key

**Additional context**
- The hosted IDP metadata _does_ have this key and is not affected.

**Comment by thijskh** [comment]
The question this raises with me is why it needs the extra key in the first place. Since the entityID is already in the array index, so why does it need to be duplicated.

**Comment by tvdijen** [comment]
Agreed, we should be able to get rid of the key everywhere and just use the array index.
